### PR TITLE
More Battle configs

### PR DESCRIPTION
**Pull request by AsparagusEduardo**

### CODE REVIEW

File: include/constants/battle_config.h [hunk 1]
```diff
@@ -53,28 +53,30 @@
...
+#define B_FELL_STINGER_STAT_RAISE   GEN_6 // In Gen7+, it raises Atk by 3 stages instead of 2 if it causes the target to faint.
+#define B_SOUND_SUBSTITUTE          GEN_6 // In Gen6+, sound moves bypass Substitute.
+#define B_TOXIC_NEVER_MISS          GEN_6 // In Gen6+, if Toxic is used by a Poison type, it will never miss.
+#define B_PAYBACK_SWITCH_BOOST      GEN_6 // In Gen5+, if the opponent switches out, Payback's damage will no longer be doubled.
+#define B_KINGS_SHIELD_LOWER_ATK    GEN_6 // In Gen7+, it lowers Atk by 1 stage instead of 2 of oponents that hit it.
 
 // Ability settings
-#define B_ABILITY_POP_UP            GEN_6 // Starting from gen5, the pokemon abilities are displayed in a pop-up, when they activate in battle.
+#define B_ABILITY_POP_UP            GEN_6 // In Gen5+, the Pokémon abilities are displayed in a pop-up, when they activate in battle.
 #define B_ABILITY_WEATHER           GEN_6 // Up to gen5 - weather induced by abilities such as Drought or Drizzle lasted till the battle's end or weather change by a move. From Gen6 onwards, weather caused by abilities lasts the same amount of turns as induced from a move.
```
Comment (DizzyEggg) [inline include/constants/battle_config.h:73]: You left B_ABILITY_WEATHER in standarization

File: src/battle_util.c [hunk 2]
```diff
@@ -5825,13 +5825,13 @@ static u32 CalcMoveBasePowerAfterModifiers(u16 move, u8 battlerAtk, u8 battlerDe
     if (gStatuses3[battlerAtk] & STATUS3_ME_FIRST)
         MulModifier(&modifier, UQ_4_12(1.5));
     if (gFieldStatuses & STATUS_FIELD_GRASSY_TERRAIN && moveType == TYPE_GRASS && IsBattlerGrounded(battlerAtk) && !(gStatuses3[battlerAtk] & STATUS3_SEMI_INVULNERABLE))
-        MulModifier(&modifier, UQ_4_12(1.5));
+        MulModifier(&modifier, (B_TERRAIN_TYPE_BOOST >= GEN_8) ? UQ_4_12(1.3) : UQ_4_12(1.5));
     if (gFieldStatuses & STATUS_FIELD_MISTY_TERRAIN && moveType == TYPE_DRAGON && IsBattlerGrounded(battlerDef) && !(gStatuses3[battlerDef] & STATUS3_SEMI_INVULNERABLE))
         MulModifier(&modifier, UQ_4_12(0.5));
```
Comment (ExoryxHana) [inline src/battle_util.c:5830]: You forgot one.

Comment (AsparagusEduardo) [reply src/battle_util.c:5830]: Misty Terrain's % wasn't changed in Gen 8, plus it doesn't boost Fairy type moves, it only decreases Dragon type moves.

File: include/constants/battle_config.h (hunk 1 above)
Comment (AsparagusEduardo) [reply include/constants/battle_config.h]: My bad ^^;

File: src/battle_util.c (hunk 2 above)
Comment (ExoryxHana) [reply src/battle_util.c:5830]: My bad.

**Pull request merged**